File: experimental/typescript/test/hello.test.ts
import { test } from 'node:test';
import assert from 'node:assert/strict';

import { HELLO, extractWorld } from '../src/hello.js';

test('extracts world', () => {
  assert.strictEqual(extractWorld(HELLO), 'world');
});

test('returns null when missing', () => {
  assert.strictEqual(extractWorld('no match here'), null);
  assert.strictEqual(extractWorld(undefined), null);
});

File: experimental/c/tests/hello_test.c
#include <assert.h>
#include <string.h>

#include "hello.h"

int main(void) {
    const char *result = get_world("hello world");
    assert(result != NULL);
    assert(strcmp(result, "world") == 0);

    assert(get_world("no match here") == NULL);
    return 0;
}
